src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -6,42 +6,85 @@
+    	  if(omf) {
+    	 // start building OMF data message body
+    	 OSIsoftServer.startOMFdataMessage();
+    	 } else {
+    		 OSIsoftServer.startBatch();
+    	 }
          ArrayList dataPoints = new ArrayList();
 
          int pointsPerTag = (maxDataPointsPerPost/OSIsoftConfig.tags.size());
          int pointsAdded = 0;
          //Build the JSON payload containing the most recent data points
          for (int tagIndex = 0; tagIndex < OSIsoftConfig.tags.size(); tagIndex++) {
             dataPoints.add(((Tag) OSIsoftConfig.tags.get(tagIndex)).getNewestDataPoints(pointsPerTag));
-
+          
+            String tagName = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagName();
+            if(omf) {
+	            if (tagIndex >0 ) {            		   
+	     		   OSIsoftServer.seperateDataMessage();
+	            }
+	            
+	            OSIsoftServer.addContainerStartToOMFdataMessage(tagName);
+            } 
+            	
             for(int dataPointIndex = 0; dataPointIndex<((ArrayList) dataPoints.get(tagIndex)).size(); dataPointIndex++)
             {
+            	
                if(((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex) != null)
                {
-                  OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
-                  pointsAdded++;
+            	   
+            	if(omf) {
+	               	String tagValue = ((Tag) OSIsoftConfig.tags.get(tagIndex)).getTagValue();
+	               	String timeStamp = ((DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex)).timestamp;
+	               	
+	            	// if there is more than one data point we need to comma separate them
+	            	   if (dataPointIndex > 0) {            		   
+	            		   OSIsoftServer.seperateDataMessage();
+	            	   }
+	            	   
+	            	   // add a data point to the OMF message body
+	            	   OSIsoftServer.addPointToOMFdataMessage(tagValue, timeStamp);            	
+            	} else {
+                	OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
+                }
+            	   pointsAdded++;
                }
                else Logger.LOG_ERR("Null data point encountered");
             }
+            if(omf) {
+            	OSIsoftServer.addContainerEndToOMFdataMessage();
+            }
+         }
 
+         if(omf) {
+        	 OSIsoftServer.endOMFdataMessage();
+         } else {
+        	 OSIsoftServer.endBatch();
          }
 
-         OSIsoftServer.endBatch();
 
          if(pointsAdded>0)
          {
             Logger.LOG_DEBUG("Sending " + pointsAdded + " points to server");
             //Post the tags to the server
-            boolean retval = OSIsoftServer.postBatch();
+            boolean retval = OSIsoftServer.postOMFBatch();

```

Comment:
Almost every line you touched in this file has a mix of tabs and spaces. also lots of trailing whitespace.